plugins/TrollManagement/class.trollmanagement.plugin.php

```diff
@@ -469,4 +493,191 @@ private function checkTroll(int $userID, array &$args) {
+
+        $configurationModel->setField([
+            'TrollManagement.PerFingerPrint.Enabled' => c('TrollManagement.PerFingerPrint.Enabled', false),
+            'TrollManagement.PerFingerPrint.MaxUserAccounts' => c('TrollManagement.PerFingerPrint.MaxUserAccounts', 5),
+        ]);
+
+        $sender->Form->setModel($configurationModel);
+
+        // If seeing the form for the first time...
+        if ($sender->Form->authenticatedPostBack() === false) {
+            $sender->Form->setData($configurationModel->Data);
+        } else {
+            $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.Enabled', 'Boolean');
+
+            if ($sender->Form->getFormValue('TrollManagement.PerFingerPrint.Enabled')) {
+                $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.MaxUserAccounts', 'Required');
+                $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.MaxUserAccounts', 'Integer');
+                $configurationModel->Validation->applyRule(
+                    'TrollManagement.PerFingerPrint.MaxUserAccounts',
+                    'validatePositiveNumber',
+                    sprintf(t('%s must be a positive number.'), t("Maximum user's accounts"))
+                );
+            }
+
+            if ($sender->Form->save()) {
+                $sender->StatusMessage = t('Your changes have been saved.');
+            }
+        }
+
+        $sender->render($this->getView('configuration.php'));
+    }
+
+    /**
+     * Add a _probable_ justification as to why a user's is on the applicant's list if there are too many user
+     * accounts using the same fingerprint.
+     *
+     * @param Controller $sender
+     * @param array $args
+     */
+    public function base_applicantInfo_handler($sender, $args) {
+        if (c('TrollManagement.PerFingerPrint.Enabled', false)) {
+            $maxSiblingAccounts = c('TrollManagement.PerFingerPrint.MaxUserAccounts');
+            $userFingerprint = val('Fingerprint', $args['User']);
+            if (!empty($userFingerprint)) {
+                if ($this->checkMaxSharedFingerprintsExceeded($userFingerprint, $maxSiblingAccounts)) {
+                    $sender->EventArguments['ApplicantMeta'][t("Fingerprint issue")] = sprintf(
+                        t("Too many accounts are using the '%s' fingerprint."),
+                        $userFingerprint
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks if the maximum amount of accounts using the same fingerprint has been reached.
+     *
+     * @param null|string $fingerprint
+     * @param int $maxSiblingAccounts
+     * @return bool
+     */
+    private function checkMaxSharedFingerprintsExceeded($fingerprint, int $maxSiblingAccounts): bool {
+        if (!is_null($fingerprint)) {
+            $sql = clone Gdn::sql();
+            $sql->reset();
+            $users = $sql
```

Comment:
Do we need to use the naked SQL object? It's probably better to use the user model, if possible.

Reply:
Alright-o. Done.